worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -1312,37 +1489,43 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
               StatusCode.PUSH_DATA_WRITE_FAIL_REPLICA
             }
           workerSource.incCounter(WorkerSource.WRITE_DATA_FAIL_COUNT)
-          writePromise.failure(new CelebornIOException(cause))
+          result(index) = cause
```

Comment:
When encounter AlreadyClosedException, we need use HARD_SPLIT status for the partition